### PR TITLE
{client/web},{ipn/ipnlocal}: replace localapi debug-web-client endpoint

### PR DESCRIPTION
**Pull request by soniaappasamy**
This change removes the existing debug-web-client localapi endpoint and replaces it with functions passed directly to the web.ServerOpts when constructing a web.ManageServerMode client.

The debug-web-client endpoint previously handled making noise requests to the control server via the /machine/webclient/ endpoints. The noise requests must be made from tailscaled, which has the noise connection open. But, now that the full client is served from tailscaled, we no longer need to proxy this request over the localapi.

Updates tailscale/corp#14335

### CODE REVIEW

File: ipn/ipnlocal/web_client.go [hunk 1]
```diff
@@ -144,3 +149,53 @@ func (b *LocalBackend) newWebClientListener(ctx context.Context, ap netip.AddrPo
...
+
+// waitWebClientAuthURL connects to the control server and blocks
+// until the associated auth URL has been completed by its user,
+// or until ctx is canceled.
+func (b *LocalBackend) waitWebClientAuthURL(ctx context.Context, id string, src tailcfg.NodeID) (*tailcfg.WebClientAuthResponse, error) {
+	return b.doWebClientNoiseRequest(ctx, id, src)
+}
+
+// doWebClientNoiseRequest handles making the "/machine/webclient"
+// noise requests to the control server for web client user auth.
```
Comment (willnorris) [inline ipn/ipnlocal/web_client.go:168]: maybe document that this method either initializes a new session, or waits for the session to be completed, based on the presence or absence of the provided `id` value ?

**Comment by soniaappasamy** [comment]
Fixed tests

**Review by willnorris — APPROVED** [review]
code looks good and works well in my local testing

File: ipn/ipnlocal/web_client.go (hunk 1 above)
Comment (soniaappasamy) [reply ipn/ipnlocal/web_client.go:168]: done!

**Pull request merged**